kong/cmd/utils/inject_directives.lua

```diff
@@ -0,0 +1,177 @@
+local conf_loader = require "kong.conf_loader"
```

Comment:
we seems to be reinventing some wheels in this file, like parsing the `argv`s and copy paste some from
`bin/kong`. Would a simpler solution like https://github.com/Kong/kong/pull/8993/files#diff-bdc96606a7bc207e18e4abce9ba3f6b43e74cf44faf4bef735c409347cb0842c work for this use case?

Reply:
And if we really want to do this respawn path, I would actually rather move `bin/kong` to be a simple shell script and handle all the lua logic in addition to this inject conf logic in a single place. I feel like the approach we are doing here is adding blocks to the CLI and making it unnecessary complicated.

Reply:
> Would a simpler solution like https://github.com/Kong/kong/pull/8993/files#diff-bdc96606a7bc207e18e4abce9ba3f6b43e74cf44faf4bef735c409347cb0842c work for this use case?

Yes, reusing the original `args` table would be simpler. I changed that part of code to do like this.

> And if we really want to do this respawn path, I would actually rather move bin/kong to be a simple shell script and handle all the lua logic in addition to this inject conf logic in a single place

After a second thought, I think `kong/cmd/init.lua` is still a more appropriate place to do the injection. Because the injection relies on the kong `conf` which in turn relies on the options `-c,--conf` and `-p,--prefix`. So the injection should be done after the args parsing. If we move this injection logic to `bin/kong` then we also need to copy those command checking and options processing to the `bin/kong`.